### DISCUSSION TITLE
[deleted]

### DISCUSSION THREAD

**Discussion by DenisIndenbom** [hufrea/byedpi#58] [Ideas]
[deleted]

**Comment by DenisIndenbom** [comment]


**Reply by decide-you** [reply]
@DenisIndenbom  Привет, с твоим конфигом грузится сайт, открываются видео, но не грузятся. Если добавить --oob 1, то все гуд. Помимо самой утилиты ты никакие настройки в конфиге браузера не подкручивал или может еще какие-то дополнительные действия?

**Reply by DenisIndenbom** [reply]
Какой у вас провайдер? МГТС? Я ничего не использую дополнительно, кроме [FoxyProxy](https://chromewebstore.google.com/detail/foxyproxy/gcknhkkoolaabfmlnjonogaaifnjlfnp). Конфиг тот же, что я и указал (`--split 1 --disorder 3+s --mod-http=h,d --auto-mode 1 --fake -1 --tlsrec 3+h --ip 127.0.0.1`). Но у меня почему-то всё работает.

**Reply by DenisIndenbom** [reply]
C вашим конфигом у меня выводятся такие ошибки/варинги:
![{DB7CAC97-D425-48C9-B21D-BD30D8D5E820}](https://github.com/user-attachments/assets/d860c4a9-fff7-4ac8-b139-ef743a22c614)
Но вроде ютуб работает.

**Reply by decide-you** [reply]
> Какой у вас провайдер? МГТС?

Ростелеком, Новосибирск. Нашел рабочий конфиг на замену `ciadpi --split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h`

**Reply by decide-you** [reply]
@DenisIndenbom Я почему написал, с вашим конфигом работают другие заблокированные ресурсы (но не грузятся видео), а с предоставленным выше только ютуб, к сожалению

**Reply by DenisIndenbom** [reply]
Попробуйте этот конфиг: `--split 1 --oob 1 --disorder 3 --mod-http=h,d --fake -1 --tlsrec 3+h --ip 127.0.0.1`. Почему-то когдя убрал `--auto none` всё заработало. Есть предположение, что в обновлении **v0.14.x** поломался параметр `--auto`.

**Reply by decide-you** [reply]
Видео по прежнему не грузятся с этим конфигом... Пока что для меня лучший вариант `--oob 1 -i 127.0.0.1`, но с ним работает только ютуб, не работает тот же твиттер

**Reply by DenisIndenbom** [reply]
К сожалению, тогда я не смогу вам помочь, т.к у меня другой провайдер.

**Reply by kiav** [reply]
> Ростелеком, Новосибирск. Нашел рабочий конфиг на замену `ciadpi --split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h`

Ростелеком, Владимир. Не работает.

**Reply by kiav** [reply]
> Попробуйте этот конфиг: `--split 1 --oob 1 --disorder 3 --mod-http=h,d --fake -1 --tlsrec 3+h --ip 127.0.0.1`. Почему-то когдя убрал `--auto none` всё заработало. Есть предположение, что в обновлении **v0.14.x** поломался параметр `--auto`.

Ростелеком, Владимир. Этого хватает на 20 секунд. Потом блокируется намертво. 
byedpi-0.14.1

**Reply by kiav** [reply]
До сегодняшнего дня работало то, что брал с https://censorship.fandom.com/ru/wiki/Byedpi
`--split 1 --oob 1 --auto r --oob 1 --auto t --fake -1 --tlsrec 1+s --auto s`

Но сегодня фсё.

**Comment by DenisIndenbom** [comment]


**Reply by X-RayGuy** [reply]
Получается прописывать после ...byedpi.exe\" -9 --blacklist ...?
что то типо такого должно выйти?
sc create "GoodbyeDPI" binPath= "\"%CD%\%_arch%\goodbyedpi.exe\" -9 --blacklist --split 1 --disoob 3 --disorder 7 --fake -1 --auto none --tlsrec 3+h --mod-http=h,d,r \"%CD%\russia-blacklist.txt\" --blacklist \"%CD%\russia-youtube.txt\"" start= "auto"
не сильно понял механизм этих флагов.ибо у меня на разных конфигурациях от -1 до -9 . Ютуб подлагивает на МГТС да и вообще отвалилось большинство сервисов. версию скачал новую 0.2.3rc3-2...
Если использовать 0.2.3rc3-2 версию и запускать сервисом -5 -e1 -f1 работает только ютуб но ролики в 720р только работают. А на самой свежей версии то вообще бывает ложится всё.Даже яндекс.....пишет нет инета

вот сейчас sc create "GoodbyeDPI" binPath= "\"%CD%\%_arch%\goodbyedpi.exe\" -2 -e1 -f1 \"%CD%\russia-blacklist.txt\" --blacklist \"%CD%\russia-youtube.txt\"" start= "auto" конфигурация и ютуб грузит на МГТС.Но трекеры условно или что либо ещё не грузит.В список добавлены все сайты,а нифига не грузит.

**Reply by X-RayGuy** [reply]
а разница есть впринципе?
просто по факту не знаю как но был на странице проекта GDPI и попал сюда)
мне главное доступность сайтов.
а запуск через какое то упрощённое приложение по сборке проектов или есть альтернатива?Не силён в программировании просто и запускать проекты подобные хз как.

**Reply by X-RayGuy** [reply]
А запускать тогда как его? не особо понял впринципе

**Comment by DescendantOfBraves** [comment]
> @decide-you 
> 
> Ростелеком, Новосибирск. Нашел рабочий конфиг на замену `ciadpi --split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h`

Сказать, что моя благодарность не знает границ — ничего не сказать. Огромное Вам спасибо, спустя две ночи мучений у меня наконец все работает идеально!

А вот набор параметров, предлагаемый тут к использованию по умолчанию, у меня, к сожалению, не заработал. На форуме GoodbyeDPI есть программа GoodCheck, которая умеет проверять конфигурации для ByeDPI в том числе. Ее вывод выглядит примерно так:
```
Strategies with 10 out of 61 successes:
--ip 127.0.0.1 --port 18081 --split 2 --disorder 3 --ttl 5 --auto=torst --timeout 3 --tlsrec 3+s

Strategies with 11 out of 61 successes:
--ip 127.0.0.1 --port 18081 --split 1 --disorder 3+s --mod-http=h,d --auto=torst --tlsrec 1+s

Strategies with 13 out of 61 successes:
--ip 127.0.0.1 --port 18081 -s1 -o1 -Ar -o1 -At - f-1 -r1+s -As

Strategies with 50 out of 61 successes:
--ip 127.0.0.1 --port 18081 --split 2 --disorder 3 --fake -1 --ttl 5
--ip 127.0.0.1 --port 18081 --split 2 --disorder 3 --ttl 5 --tlsrec 3+s --auto=torst --timeout 3
```
Где `50 out of 61` — количество сайтов из списка заблокированных, которые открылись. Работает пока не идеально — в частности, вот эти `50 out of 61` это плохие параметры, так на первом все сильно тормозит (зато хотя бы работает), а второй летает как ракета во всем, кроме самих видео, которые в 100% случаев не загружает (а на других сайтах пишет ошибку, связанную с TLS или SSL). Если бы можно было совместить проверку, которую делает эта программа, с программой ByeDPI, чтобы она автоматически предлагала параметры для имеющегося окружения, вот это было бы здорово.

А вообще, с этими блокировками у меня такое ощущение, будто Родину отбирают...

**Reply by NSKVooDoo** [reply]
> Будьте добры поделиться ссылкой на репозиторий GoodCheck. Интересно узнать об этой утилите чуть подробнее.

https://ntc.party/t/goodcheck-%D0%B1%D0%BB%D0%BE%D0%BA%D1%87%D0%B5%D0%BA-%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%82-%D0%B4%D0%BB%D1%8F-goodbyedpi-zapret-byedpi/10880
Не уверен, что есть публичный репозиторий, но поскольку это просто скрипт - исходники не скомпилены и доступны для анализа.

**Comment by poiNt3D** [comment]
Сегодня перестало работать. Уже вариантов 10 перебрал.

**Reply by DenisIndenbom** [reply]
Да, я тоже заметил. Я подобрал новый конфиг, который работает у МГТС, но не уверен насчёт других провайдеров.

**Reply by neljkeee** [reply]
Мог бы поделиться? Думаю многим будет крайне полезно.

**Reply by DenisIndenbom** [reply]
Я написал новый вариант в сообщение выше, но уже тоже не работает.

**Reply by NSKVooDoo** [reply]
Тоже Ростелеком, Новосибирск. Помог конфиг `--split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com`, хотя при этом GoodCheck его забраковал:
```
Strategies with 0 out of 30 successes:
--ip 127.0.0.1 --port 18081 --split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com
```

**Reply by kiav** [reply]
> --split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com

Ростелеком, Владимир. Заработало, но не сразу. Подождать пришлось с минуту, пока браузер думал.
При тыке на произвольное место видео (для пропуска рекламы) может и не работать (у меня не работает). Проходит просмотр от начала и не прерываясь до конца.

**Reply by GennadijK1987** [reply]
> > --split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com
> 
> Ростелеком, Владимир. Заработало, но не сразу. Подождать пришлось с минуту, пока брайзер думал.

Эти параметры нужно вводить в Goodbye dpi ? Если да , то в каком имено файле ? Можно ли использовать в 1_russia_blacklist_YOUTUBE ? Или этот параметр предназначен для другой программы ? Или это для BYE DPI ?

**Reply by DenisIndenbom** [reply]
Нет, эти параметры для ByeDPI. Вы ошиблись репозиторием.

**Reply by kiav** [reply]
> --split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com

А кончилось щастье. Хватило на пару часов.
А еще через 2 часа снова заработало. Не знаю что происходит.

**Reply by antonnagorniy** [reply]
Ростелеком северо-запад - старый конфиг сегодня отвалился, но, юпуп, всякие граммы и трекеры заработали с параметрами от товарища в треде выше:
--split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h

**Reply by wolfer-uwu** [reply]
> Тоже Ростелеком, Новосибирск. Помог конфиг `--split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com`, хотя при этом GoodCheck его забраковал:
> 
> ```
> Strategies with 0 out of 30 successes:
> --ip 127.0.0.1 --port 18081 --split 2 --fake 0+e --ttl 5 --tls-sni=www.google.com
> ```

РТК, Иркутск, работало 15 минут, как у ваc с этим конфигом сейчас?

**Reply by NSKVooDoo** [reply]
> как у ваc с этим конфигом сейчас?  

Почти всё ок. Длинные видео нормально работают, хотя вначале стартуют с задержкой в секунд пять. А вот шортсы не юзабельно тормозят.

**Reply by aquadim** [reply]
> Ростелеком северо-запад - старый конфиг сегодня отвалился, но, юпуп, всякие граммы и трекеры заработали с параметрами от товарища в треде выше: --split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h

Ростелеком, Волго-Вятский, работает но надо подождать пока браузер подумает

**Reply by poiNt3D** [reply]
А кто-нибудь может поделиться каким образом вы подбираете конфиг? Просто перебор всех возможных опций и значений, или есть какая-то логика?

**Reply by kiav** [reply]
> --split 1 --oob 1 --mod-http=h,d --auto none --fake -1 --tlsrec 3+h

Ну а Ростелеком, Владимир не работает. И ожидание ничего не дает.

> А кто-нибудь может поделиться каким образом вы подбираете конфиг? Просто перебор всех возможных опций и значений, или есть какая-то логика?

Для этого нужно хорошо понимать работу протокола. Думаю, многие здесь (как и я) просто собирают "волшебные строки". Кто-то тут писал про утилиту GoodCheck. Может она что дает. Не разбирался.

**Reply by rezawq** [reply]
```--split 2 --disorder 3 --fake -1 --ttl 5```

у меня так норм. Ростелеком пропускает

**Reply by kiav** [reply]
> Предполагаю, что после изменения в работе DPI, столь большой поток данных стало легче детектить даже при всех манипуляций с пакетами.

Мне тоже так кажется. Ибо каждая новая "волшебная строка" работает нормально несколько часов. Либо система быстро адаптируется, либо они сидят в этой и таких же темам и смотрять что придумал народ.



> Остаётся перебирать или ждать решения от более находчивых.

Думаю нужно "таблетку" посерьезнее изобретать. Здесь нужны случайные стратегии "одурачивания" перебираемые тоже случайно.

**Reply by poiNt3D** [reply]
Будем надеяться, что в итоге детектирование станет настолько сложным, что будет давать false-positive и начнут ломаться вещи, которые не должны по их мнению ломаться.

**Reply by trevarj** [reply]
Кажется что [у запрета есть конфиг](https://github.com/bol-van/zapret/discussions/200), но я не знаю как правильно перевести на byedpi, именно [этот парам](https://github.com/bol-van/zapret/blob/2cd6db3ba5ac2fa1494bed1c1903bc3531c76bc5/docs/readme.txt#L251C2-L251C28) 😞

**Comment by zimbabwe303** [comment]
Никак не мог подобрать строку, чтобы работал FreeTube (на Linux); в конце концов внезапно обнаружил, что сочетание `--md5sig` и некоего небольшого значения `--buf-size` (находится опытным путём) помогает. Видео стартуют мгновенно.

Моя минимальная работающая строка: `-s1 -q1 -f-1 -S -b661 -As`

Число 661 - для меня минимальное при котором грузятся сайты, почему оно такое - понятия не имею.

**Reply by alekseyaz** [reply]
Спасибо тебе дружище, дай я тебя расцелую, все заработало!

**Reply by edvejak** [reply]
> Никак не мог подобрать строку, чтобы работал FreeTube (на Linux); в конце концов внезапно обнаружил, что сочетание `--md5sig` и некоего небольшого значения `--buf-size` (находится опытным путём) помогает. Видео стартуют мгновенно.
> 
> Моя минимальная работающая строка: `-s1 -q1 -f-1 -S -b661 -As`
> 
> Число 661 - для меня минимальное при котором грузятся сайты, почему оно такое - понятия не имею.

It's very strange, but it works! There's no -q parameter in cia for linux

**Reply by goberghen** [reply]
> There's no -q parameter in cia for linux

There is
`    -q, --disoob <n[+s]>      Split and send reverse order as OOB data`

**Reply by edvejak** [reply]
> > There's no -q parameter in cia for linux
> 
> There is ` -q, --disoob <n[+s]> Split and send reverse order as OOB data`

$ /usr/local/bin/ciadpi-x86_64 --help
    -i, --ip, <ip>            Listening IP, default 0.0.0.0
    -p, --port <num>          Listening port, default 1080
    -c, --max-conn <count>    Connection count limit, default 512
    -N, --no-domain           Deny domain resolving
    -U, --no-udp              Deny UDP association
    -I  --conn-ip <ip>        Connection binded IP, default ::
    -b, --buf-size <size>     Buffer size, default 16384
    -x, --debug <level>       Print logs, 0, 1 or 2
    -g, --def-ttl <num>       TTL for all outgoing connections
    -F, --tfo                 Enable TCP Fast Open
    -A, --auto[=t,r,c,s,a,n]  Try desync params after this option
                              Detect: torst,redirect,cl_err,sid_inv,alert,none
    -u, --cache-ttl <sec>     Lifetime of cached desync params for IP
    -T, --timeout <sec>       Timeout waiting for response, after which trigger auto
    -K, --proto <t,h,u>       Protocol whitelist: tls,http,udp
    -H, --hosts <file|:str>   Hosts whitelist, filename or :string
    -V, --pf <port[-portr]>   Ports range whitelist
    -s, --split <n[+s]>       Split packet at n
                              +s - add SNI offset
                              +h - add HTTP Host offset
    -d, --disorder <n[+s]>    Split and send reverse order
    -o, --oob <n[+s]>         Split and send as OOB data
    -f, --fake <n[+s]>        Split and send fake packet
    -t, --ttl <num>           TTL of fake packets, default 8
    -k, --ip-opt[=f|:str]     IP options of fake packets
    -S, --md5sig              Add MD5 Signature option for fake packets
    -l, --fake-data <f|:str>  Set custom fake packet
    -n, --tls-sni <str>       Change SNI in fake ClientHello
    -e, --oob-data <f|:str>   Set custom OOB data, filename or :string
    -M, --mod-http <h,d,r>    Modify HTTP: hcsmix,dcsmix,rmspace
    -r, --tlsrec <n[+s]>      Make TLS record at position
    -a, --udp-fake <count>    UDP fakes count, default 0

**Reply by zimbabwe303** [reply]
@edvejak You're obviously using a very old version, download the [latest](https://github.com/hufrea/byedpi/releases/latest) one. `-q` was introduced in v0.13 on August 23, and a lot more was added since then.

**Comment by zimbabwe303** [comment]
С сегодняшними изменениями в master-ветке (версия 1.8) строка `-s1 -q1 -Y -At -T5 -b1000 -S -f-1 -r1+sm -As` для меня прекрасно работает везде, включая [FreeTube](https://github.com/FreeTubeApp/FreeTube) на Linux (версия из Flatpak).

**Reply by vehametra** [reply]
_Call the post desync func only once_
 до него эта строка ломала  большинство сайтов и  тормозило, но dpi  пробивало. Сейчас успешно работает.
 Ютуб у нас разблокировали, проверял на твитере.
 Работоспособность подтверждаю, схороню. 
 Интересно, насколько важен размер буффера, и почему маленький.
`srss/byedpi$ ./ciadpi  -s1 -q1 -Y -At -T5 -b1000000 -S -f-1 -r1+sm -As -x1  `

так у меня тоже работатет. Но проверял опятьже на твиттере.

**Reply by edvejak** [reply]
> С сегодняшними изменениями в master-ветке (версия 1.8) строка `-s1 -q1 -Y -At -T5 -b1000 -S -f-1 -r1+sm -As` для меня прекрасно работает везде, включая [FreeTube](https://github.com/FreeTubeApp/FreeTube) на Linux (версия из Flatpak).

Where did you find 1.8 version of master branch? I can't see it here

**Reply by zimbabwe303** [reply]
No release was made yet but the version number was updated in the code, just use either `git clone https://github.com/hufrea/byedpi.git` or https://github.com/hufrea/byedpi/archive/refs/heads/main.zip

**Reply by edvejak** [reply]
> No release was made yet but the version number was updated in the code, just use either `git clone https://github.com/hufrea/byedpi.git` or https://github.com/hufrea/byedpi/archive/refs/heads/main.zip
Don't work for me at all. v0.14.1 works like a charm: Lubuntu 24.04.1

**Comment by remalex123** [comment]
-s1 -q1 -Y -At -T5 -b1000 -S -f-1 -r1+sm -As
работает!

**Reply by zimbabwe303** [reply]
По прежнему не могу одолеть остановку потока, происходящую через какое-то случайное время. Не знаю, заметна ли эта проблема в браузере, но для yt-dl и особенно FreeTube она точно есть. У кого-нибудь есть идеи, что можно подкрутить? Вот если бы была возможность либо автоматического реконнекта в случае остановки потока, или же искусственного ограничения скорости (есть подозрение, что DPI реагирует на высокоскоростные потоки данных, как признак того, что его обдурили).

**Reply by xtzorig** [reply]
Может попробовать yt-dlp? 
не обнаружил падения скорости

**Reply by zimbabwe303** [reply]
Пардон, это и есть yt-dlp: у меня просто скриптик называется yt-dl, который запускает yt-dlp сразу с аргументом `--proxy=socks5h://127.0.0.1:1080`, вот я и привык к такому названию. В любом случае, проблема такая, что оно может докачать и нормально без единой ошибки, а может и 15 раз retry сделать: зависит, прежде всего, от размера видео. Чем дольше висит коннект со скоростным потоком данных, тем чаще ТСПУ его внезапно строттливает до нуля. Причём не плавно, а резко - и это не дисконнект, а именно троттлинг, т.к. в byedpi просто выходит таймаут (для этого и стоит аргумент `-T5` - 5 секунд). Для коротких видео вообще почти всё всегда нормально. Ещё: если снова качать не сразу, а спустя какое-то время, то опять поток сначала идёт нормально. Тут налицо какой-то статистический анализ.
